### PR TITLE
Automatic addition of all current keyboards during installation

### PR DESCRIPTION
**Pull request by joonhoekim**
Changes:
- Added Script file `/scripts/update_conf_with_all_keyboards.sh`
- During `make install`, if the user agrees, the script modifies `~/.config/touchcursor/touchcursor.conf`
- The default option is 'N'. (To prevent accidental config changes.)
- Configuration is updated based on CONFIGPATH and CONFIG variables 

Script
- Only the [Keyboard] section is modified.
- If the script is run, it will create a backup of the conf file as `~/.config/touchcursor/touchcursor.conf.bak`

Requirements:
- Bash is required for script execution.

image

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -75,12 +75,22 @@ install:
...
 
+	@echo "# Do you want to add the configuration with current all keyboards?(y/N)"
+	@read -r answer; \
+	if [ "$$answer" = "y" ] || [ "$$answer" = "Y" ]; then \
+		echo "\nUpdating configuration with all keyboards..."; \
+		bash ./scripts/update_conf_with_all_keyboards.sh $(CONFIGPATH)/$(config); \
+	else \
+		echo "\nSkipping update conf"; \
+	fi
+	@echo ""
```
Comment (donniebreve) [inline Makefile:92]: Suggestion on the grammar here:
```suggestion
	@echo "# Do you want to add all currently connected keyboards to your configuration? (y/N)"
	@read -r answer; \
	if [ "$$answer" = "y" ] || [ "$$answer" = "Y" ]; then \
		echo "\nUpdating configuration..."; \
		bash ./scripts/update_conf_with_all_keyboards.sh $(CONFIGPATH)/$(config); \
	fi
	@echo ""
```

Any reason for the capital `N` in the message (at `(y/N)`)?

File: scripts/update_conf_with_all_keyboards.sh [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+if [ $# -eq 0 ]; then
+    echo "Error: No configuration file specified."
+    echo "Usage: $0 <path_to_config_file>"
+    exit 1
+fi
+
+CONFIG_FILE="$1"
+
+# Set permissions to allow read and write for all users
+chmod 644 "$CONFIG_FILE"
```
Comment (donniebreve) [inline scripts/update_conf_with_all_keyboards.sh:13]:
```suggestion
# Set permissions to allow write for owner and read for others
chmod 644 "$CONFIG_FILE"
```

File: scripts/update_conf_with_all_keyboards.sh [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+# Check if the temporary file is not empty and is readable
+if [ -s "$TEMP_FILE" ] && [ -r "$TEMP_FILE" ]; then
+    # Create a backup of the original file
+    cp "$CONFIG_FILE" "${CONFIG_FILE}.bak"
+    
+    # Replace the original file with the modified content
+    mv "$TEMP_FILE" "$CONFIG_FILE"
+    chmod 644 "$CONFIG_FILE"
+    
+    echo "Global configuration in $CONFIG_FILE has been updated."
```
Comment (donniebreve) [inline scripts/update_conf_with_all_keyboards.sh:69]:
```suggestion
    echo "Configuration ($CONFIG_FILE) has been updated."
```

**Review by donniebreve — CHANGES_REQUESTED** [review]
Looks good so far :rocket:

File: Makefile (hunk 1 above)
Comment (joonhoekim) [reply Makefile:92]: I totally agree with the grammar and sentence feedback. As a non-native English speaker, this kind of feedback is extremely valuable to me.

The reason 'N' is capitalized is to indicate to users that the default value is 'N'.

Comment (donniebreve) [reply Makefile]: Thanks for the explanation. This makes sense to me.

File: Makefile [hunk 4]
```diff
@@ -75,12 +75,20 @@ install:
...
+	
 	@echo "# Copying default configuration file to $(CONFIGPATH)/$(config)"
 	mkdir --parents $(CONFIGPATH)
 	-cp --no-clobber $(config) $(CONFIGPATH)
 	@echo ""
 
+	@echo "# Do you want to add all currently connected keyboards to your configuration? (y/N)"
+	@read -r answer; \
+	if [ "$$answer" = "y" ] || [ "$$answer" = "Y" ]; then \
+		echo "\nUpdating configuration..."; \
```
Comment (donniebreve) [inline Makefile:87]: This should be
```suggestion
		echo ""; \
		echo "Updating configuration..."; \
```
_Matches other new lines in this file_

or

```suggestion
		echo -e "\nUpdating configuration..."; \
```

The \n showed up in the output when I ran it.

Comment (joonhoekim) [reply Makefile]: Checked! I didn't catch print convention. I commited the first version that using `echo ""`. Thanks!

**Review by donniebreve — APPROVED** [review]
:rocket: Nice job!

**Pull request merged**